lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -115,7 +116,7 @@ def eval_block
 
         def verify_message
           return true if @expected_message.nil?
-          values_match?(@expected_message, @actual_error.message)
+          values_match?(@expected_message, @actual_error.message.to_s)
```

Comment:
So this change seems (to me at least) to be completely unrelated to the change of this PR.  I guess your change surfaced a bug in the `raise_error` matcher that this fixes?  If that's the case, can you add a spec to `raise_error_spec.rb` to specifically cover this behavior with an example rather than implicitly relying upon the composable specs?